### PR TITLE
Eliminate em

### PR DESCRIPTION
**Pull request by rgiordan**
I'm not sure if this will work with Travis before I tag a new version of SloanDigitalSkySurvey.

### CODE REVIEW

**Comment by rgiordan** [comment]
I tagged a new version at https://github.com/JuliaLang/METADATA.jl/pull/4740 -- I'll just put this in REQUIRE after it's approved.

**Comment by rgiordan** [comment]
These tests are failing because the stamps (and so the unit tests) all have three PSF components baked into them, which is going to make it a little tricky to move everything to two components.  :(

**Comment by jeff-regier** [comment]
@rgiordan   Do you have major objections to merging the SloanDigitalSkySurvey package back into the Celeste package? Keeping them separate doesn't make a lot of sense as an abstraction---Celeste is still very much tied to SDSS. More concretely, having to tag new versions of SloanDigitalSkySurvey makes even small changes slow. Also, old versions of Celeste don't run because they checkout the newest version of  SloanDigitalSkySurvey (could have been avoided it we'd used version numbers in the require file, but easy to forget to update that) and the travis tests for SloanDigitalSkySurvey happen with a different configuration (again, easily fixed, but do we want the flexibility to get this wrong?)

**Comment by kbarbary** [comment]
:+1: to merging back into Celeste.

**Comment by rgiordan** [comment]
Sounds good, let's merge them.

**Comment by kbarbary** [comment]
What's up with the `v0.0.2` SDSS requirement?

Practically, for merging into Celeste, it would be great if Ryan could add the `PSF` and `WCSUtils` submodules into Celeste but leave the `SDSS` submodule in `SloanDigitalSkySurvey`. In a branch I'm working on, I remove the dependence on `SloanDigitalSkySurvey.SDSS`. Once that is done, I would then finally remove the requirement on `SloanDigitalSkySurvey`.

**Comment by rgiordan** [comment]
> What's up with the v0.0.2 SDSS requirement?
> 
> It's the first tagged version with the least squares fit and without
> GaussianMixtures.
> 
> Practically, for merging into Celeste, it would be great if Ryan could add
> the PSF and WCSUtils submodules into Celeste but leave the SDSS submodule
> in SloanDigitalSkySurvey. In a branch I'm working on, I remove the
> dependence on SloanDigitalSkySurvey.SDSS. Once that is done, I would then
> finally remove the requirement on SloanDigitalSkySurvey.
> 
> Maybe while we're all iterating quickly on the same code base this is not
> the time to merge these packages.  After this change, if we really need to
> change things in SDSS we can just re-implement them in Celeste.  I think
> removing dependencies is the only thing that really forces us to make
> synced changes with SloanDigitalSkySurvey.jl.

**Comment by rgiordan** [comment]
> What's up with the v0.0.2 SDSS requirement?
> 
> Oops, that was not the right version number.  Sorry.

**Comment by rgiordan** [comment]
@kbarbary , @jeff-regier , @rcthomas , @kpamnany , this is passing tests both on the nightly and v0.4, so I'm going to merge it.  However, let me warn you that the current PSF fitting is really slow.  I'm working now to re-make the stamps with only two PSF components, speeds things up a lot.  But to get everyone going as quickly as possible on v0.5 I'll merge this first.

**Comment by kpamnany** [comment]
I'm having difficulties now:

```
ERROR: Base.Pkg.PkgError("fixed packages introduce conflicting requirements for SloanDigitalSkySurvey: \n         Celeste requires versions [0.1.1,?) [none of the available versions can satisfy this requirement]\n       available versions are 0.0.1, 0.0.2, 0.0.3 and 0.1.0")
 in check_requirements(::Dict{ByteString,Base.Pkg.Types.VersionSet}, ::Dict{ByteString,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{ByteString,Base.Pkg.Types.Fixed}) at ./pkg/query.jl:111
 in resolve(::Dict{ByteString,Base.Pkg.Types.VersionSet}, ::Dict{ByteString,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{ByteString,Tuple{VersionNumber,Bool}}, ::Dict{ByteString,Base.Pkg.Types.Fixed}, ::Dict{ByteString,VersionNumber}) at ./pkg/entry.jl:429
 in resolve(::Dict{ByteString,Base.Pkg.Types.VersionSet}, ::Dict{ByteString,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{ByteString,Tuple{VersionNumber,Bool}}) at ./pkg/entry.jl:414
 in (::Base.Pkg.Entry.##22#24{Base.LibGit2.GitRepo,ASCIIString})(::Base.LibGit2.GitRepo) at ./pkg/entry.jl:414
 in transact(::Base.Pkg.Entry.##22#24{Base.LibGit2.GitRepo,ASCIIString}, ::Base.LibGit2.GitRepo) at ./libgit2.jl:493
 in (::Base.Pkg.Entry.##21#23{ASCIIString})(::Base.LibGit2.GitRepo) at ./pkg/entry.jl:255
 in with(::Base.Pkg.Entry.##21#23{ASCIIString}, ::Base.LibGit2.GitRepo) at ./libgit2/types.jl:481
 in free(::ASCIIString) at ./pkg/entry.jl:247
 [inlined code] from ./boot.jl:331
 in (::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#free,Tuple{ASCIIString}})() at ./pkg/dir.jl:31
 in cd(::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#free,Tuple{ASCIIString}}, ::UTF8String) at ./file.jl:48
 in #cd#1(::Array{Any,1}, ::Any, ::Any, ::ASCIIString, ::Vararg{ASCIIString}) at ./pkg/dir.jl:31
 [inlined code] from ./boot.jl:331
 in free(::ASCIIString) at ./pkg.jl:160
 in eval(::Module, ::Any) at ./boot.jl:267
```

How should I clear this?

**Comment by rgiordan** [comment]
I presume you've run pkg.update(), right?  Just checking before looking
into it.

-- Ryan.  Sent from a phone.  Nevertheless, all typos are deliberate.

**Comment by kpamnany** [comment]
I did, but I get the following warning on Linux:

```
WARNING: SloanDigitalSkySurvey is fixed at 0.1.0+ conflicting with requirement for Celeste: [0.1.1,?)
```

And the following error on OS X:

```
ERROR: Base.Pkg.PkgError("unknown package WCS required by Celeste")
 in requirements(::Dict{ByteString,Base.Pkg.Types.VersionSet}, ::Dict{ByteString,Base.Pkg.Types.Fixed}, ::Dict{ByteString,Dict{VersionNumber,Base.Pkg.Types.Available}}) at ./pkg/query.jl:11
 in resolve(::Dict{ByteString,Base.Pkg.Types.VersionSet}, ::Dict{ByteString,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{ByteString,Tuple{VersionNumber,Bool}}, ::Dict{ByteString,Base.Pkg.Types.Fixed}, ::Dict{ByteString,VersionNumber}) at ./pkg/entry.jl:414 (repeats 2 times)
 in (::Base.Pkg.Entry.##18#20{ASCIIString,ASCIIString,Bool,Bool})(::Base.LibGit2.GitRepo) at ./pkg/entry.jl:414
 in transact(::Base.Pkg.Entry.##18#20{ASCIIString,ASCIIString,Bool,Bool}, ::Base.LibGit2.GitRepo) at ./libgit2.jl:493
 [inlined code] from ./pkg/entry.jl:228
 in with(::Base.Pkg.Entry.##17#19{ASCIIString,ASCIIString,Bool,Bool}, ::Base.LibGit2.GitRepo) at ./libgit2/types.jl:481
 in checkout(::ASCIIString, ::ASCIIString, ::Bool, ::Bool) at ./pkg/entry.jl:227
 [inlined code] from ./boot.jl:331
 in (::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#checkout,Tuple{ASCIIString,ASCIIString,Bool,Bool}})() at ./pkg/dir.jl:31
 in cd(::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#checkout,Tuple{ASCIIString,ASCIIString,Bool,Bool}}, ::UTF8String) at ./file.jl:47
 in #cd#1(::Array{Any,1}, ::Any, ::Any, ::ASCIIString, ::Vararg{Any}) at ./pkg/dir.jl:31
 [inlined code] from ./boot.jl:331
 in #checkout#1(::Array{Any,1}, ::Base.Pkg.#checkout, ::ASCIIString) at ./pkg.jl:147
 in eval(::Module, ::Any) at ./boot.jl:267
```

**Comment by rgiordan** [comment]
Have you pinned the sdss package?  If not, I'm not sure why you're getting
that message.

-- Ryan.  Sent from a phone.  Nevertheless, all typos are deliberate.

**Comment by kpamnany** [comment]
I don't recall pinning it. Tried to `Pkg.free()` it, but ... hold on, I'm doing a reset of `~/.julia/v0.5` and will try again.

**Comment by kpamnany** [comment]
Okay. All packages are up to date except for `Optim` and `DataFrames` which are on master as `.travis.yml` suggests. And same error:

```
kpamnany@cori06:~/.julia/v0.5/Celeste> ./bin/celeste.jl infer test/data 3900 6 269 --part 1/100
WARNING: New definition
    write(GZip.GZipStream, Array{#T<:Any, N<:Any}) at /global/homes/k/kpamnany/.julia/v0.5/GZip/src/GZip.jl:456
is ambiguous with:
    write(Base.IO, Array{UInt8, N<:Any}) at io.jl:154.
To fix, define
    write(GZip.GZipStream, Array{UInt8, N<:Any})
before the new definition.
WARNING: local variable NumType conflicts with a static parameter in #call#6 at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/bivariate_normals.jl:27.
Reading band 1 image data... masking image... reading psf... ERROR: LoadError: Base.LinAlg.PosDefException(46918222741504)
 [inlined code] from ./linalg/triangular.jl:17
 in chol!(::Array{Float64,2}, ::Type{UpperTriangular{T,S<:AbstractArray{T,2}}}) at ./linalg/cholesky.jl:28
 [inlined code] from ./linalg/cholesky.jl:34
 in wrap_parameters(::Array{Array{Float64,1},1}, ::Array{Array{Float64,2},1}, ::Array{Float64,1}) at /global/homes/k/kpamnany/.julia/v0.5/SloanDigitalSkySurvey/src/PSF.jl:46
 in #fit_psf_gaussians_least_squares#1(::Array{Float64,1}, ::Float64, ::Int64, ::Bool, ::Int64, ::Symbol, ::Any, ::Array{Float64,2}) at /global/homes/k/kpamnany/.julia/v0.5/SloanDigitalSkySurvey/src/PSF.jl:180
 [inlined code] from ./boot.jl:331
 in fit_raw_psf_for_celeste(::Array{Float64,2}) at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/SkyImages.jl:481
 in #load_sdss_blob#5(::Set{ASCIIString}, ::Any, ::UTF8String, ::ASCIIString, ::ASCIIString, ::ASCIIString) at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/SkyImages.jl:420
 [inlined code] from ./dict.jl:496
 in infer(::UTF8String, ::ASCIIString, ::ASCIIString, ::ASCIIString, ::UTF8String, ::Int64, ::Int64) at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/api.jl:62
 in main() at /global/u1/k/kpamnany/.julia/v0.5/Celeste/bin/celeste.jl:70
 in include(::ASCIIString) at ./boot.jl:264
 in include_from_node1(::UTF8String) at ./loading.jl:417
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading /global/u1/k/kpamnany/.julia/v0.5/Celeste/bin/celeste.jl, in expression starting on line 82
```

**Comment by rgiordan** [comment]
Ok, so the good news is your package conflict is resolved.  It appears to
be having some trouble actually fitting the PSF of the image you're
loading.  Let me try to reproduce on my machine.  If I can't, then perhaps
you have some kind of corrupted image file?

**Comment by kpamnany** [comment]
Do I erase `data/test/*fit*` and `make` in that directory again to clean the images?

**Comment by rgiordan** [comment]
No, don't erase all the fits files in data/test.  Unfortunately, make is
only set up to only download a subset of those needed for Celeste.  Delete
the fits files corresponding to the run / camcol / field you're looking at.
 (Or you can just use bin/download_fits_files.py, which does the same thing
a little more transparently and which will overwrite your old files when
you run it.)

**Comment by rgiordan** [comment]
Ok, I don't get this error on my version of julia5, at least.  I last
compiled it a while ago, but it's a little hard a priori to imagine a
change in julia leading to a degenerate PSF component.  I tentatively
suspect your image files.

**Comment by kpamnany** [comment]
Unfortunately:

```
kpamnany@cori06:~/.julia/v0.5/Celeste> bin/download_fits_files.py 
Traceback (most recent call last):
  File "bin/download_fits_files.py", line 7, in <module>
    from tractor.sdss import *
ImportError: No module named tractor.sdss
```

How do I get this module?

**Comment by kpamnany** [comment]
Is 3900 6 269 (from the celeste.jl command line) the run/camcol/field?

**Comment by rgiordan** [comment]
Yes

-- Ryan.  Sent from a phone.  Nevertheless, all typos are deliberate.

**Comment by kpamnany** [comment]
Okay. I'm doing all this on Cori. I erased the files that `test/data/Makefile` downloads and had it download them again. Here's my package status:

```
julia> Pkg.status()
1 required packages:
 - GaussianMixtures              0.0.11+            master
34 additional packages:
 - ArrayViews                    0.6.4
 - BinDeps                       0.3.21
 - Blosc                         0.1.4
 - Calculus                      0.1.14
 - Celeste                       0.1.0+             master
 - Clustering                    0.5.0
 - Compat                        0.7.12
 - DataArrays                    0.2.20
 - DataFrames                    0.6.10+            master
 - Distances                     0.3.0
 - Distributions                 0.8.10
 - DocOpt                        0.1.2
 - Docile                        0.5.23
 - Dtree                         0.0.0-             master (unregistered, dirty)
 - DualNumbers                   0.2.2
 - FITSIO                        0.8.2
 - FileIO                        0.0.4
 - ForwardDiff                   0.1.6
 - GZip                          0.2.18
 - Grid                          0.4.0
 - HDF5                          0.5.8
 - JLD                           0.5.9
 - NaNMath                       0.2.0
 - Optim                         0.4.4+             master
 - PDMats                        0.4.0
 - Reexport                      0.0.3
 - SHA                           0.1.2
 - SloanDigitalSkySurvey         0.1.1
 - SortingAlgorithms             0.0.6
 - StatsBase                     0.8.0
 - StatsFuns                     0.2.0
 - URIParser                     0.1.3
 - WCS                           0.1.1
 - WoodburyMatrices              0.1.5
```

And the error is the same:

```
kpamnany@cori06:~/.julia/v0.5/Celeste> bin/celeste.jl infer test/data 3900 6 269 --part 1/100
WARNING: New definition 
    write(GZip.GZipStream, Array{#T<:Any, N<:Any}) at /global/homes/k/kpamnany/.julia/v0.5/GZip/src/GZip.jl:456
is ambiguous with: 
    write(Base.IO, Array{UInt8, N<:Any}) at io.jl:154.
To fix, define 
    write(GZip.GZipStream, Array{UInt8, N<:Any})
before the new definition.
WARNING: local variable NumType conflicts with a static parameter in #call#6 at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/bivariate_normals.jl:27.
Reading band 1 image data... masking image... reading psf... ERROR: LoadError: Base.LinAlg.PosDefException(46909632806912)
 [inlined code] from ./linalg/triangular.jl:17
 in chol!(::Array{Float64,2}, ::Type{UpperTriangular{T,S<:AbstractArray{T,2}}}) at ./linalg/cholesky.jl:28
 [inlined code] from ./linalg/cholesky.jl:34
 in wrap_parameters(::Array{Array{Float64,1},1}, ::Array{Array{Float64,2},1}, ::Array{Float64,1}) at /global/homes/k/kpamnany/.julia/v0.5/SloanDigitalSkySurvey/src/PSF.jl:46
 in #fit_psf_gaussians_least_squares#1(::Array{Float64,1}, ::Float64, ::Int64, ::Bool, ::Int64, ::Symbol, ::Any, ::Array{Float64,2}) at /global/homes/k/kpamnany/.julia/v0.5/SloanDigitalSkySurvey/src/PSF.jl:180
 [inlined code] from ./boot.jl:331
 in fit_raw_psf_for_celeste(::Array{Float64,2}) at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/SkyImages.jl:481
 in #load_sdss_blob#5(::Set{ASCIIString}, ::Any, ::UTF8String, ::ASCIIString, ::ASCIIString, ::ASCIIString) at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/SkyImages.jl:420
 [inlined code] from ./dict.jl:496
 in infer(::UTF8String, ::ASCIIString, ::ASCIIString, ::ASCIIString, ::UTF8String, ::Int64, ::Int64) at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/api.jl:62
 in main() at /global/u1/k/kpamnany/.julia/v0.5/Celeste/bin/celeste.jl:68
 in include(::ASCIIString) at ./boot.jl:264
 in include_from_node1(::UTF8String) at ./loading.jl:417
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading /global/u1/k/kpamnany/.julia/v0.5/Celeste/bin/celeste.jl, in expression starting on line 80
```

**Comment by jeff-regier** [comment]
This is working for me now on Cori, when I start with an empty .julia
directory. I'm using a just-downloaded version of Julia nightly, not the
version Rollin built, but that probably doesn't make the difference.

**Comment by jeff-regier** [comment]
Except that it does! I'm getting the PosDefException too when I use the from-source version of Julia (on Cori), but not when I use the downloaded nightly binary version (on Cori). The  difference could be either because of how the two were built, or because the currently nightly is a slightly more recent version.

**Comment by rgiordan** [comment]
For debugging, please check out the branch debug_for_kiran, run
dev/debug_for_kiran.jl, and send the output?

Ryan

**Comment by rcthomas** [comment]
I can test build the nightly myself on Cori and we can see if the issue persists.  Got a commit?

**Comment by kpamnany** [comment]
We can't use the nightly binary because it is built without threads. We have to use our own build.

**Comment by rgiordan** [comment]
Figuring this out may be more trouble than using a more stable
unconstraining parameterization for the covariance matrix than a Cholesky
decomposition.  Doing that may require more attention than I can pay during
a meeting, though, but I'll try.

**Comment by kpamnany** [comment]
Output from the debug script is at https://gist.github.com/kpamnany/9c5f30e9d3f654194512#file-gistfile1-txt.

I'll try a more recent Julia commit myself.

**Comment by kpamnany** [comment]
Just built current Julia master and am seeing ~~the same~~ a different error:

```
kpamnany@cori05:~/.julia/v0.5/Celeste> bin/celeste.jl infer test/data 3900 6 269 --part 1/100
WARNING: New definition 
    write(GZip.GZipStream, Array{#T<:Any, N<:Any}) at /global/homes/k/kpamnany/.julia/v0.5/GZip/src/GZip.jl:456
is ambiguous with: 
    write(Base.IO, Array{UInt8, N<:Any}) at io.jl:154.
To fix, define 
    write(GZip.GZipStream, Array{UInt8, N<:Any})
before the new definition.
WARNING: local variable NumType conflicts with a static parameter in #call#6 at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/bivariate_normals.jl:27.
Reading band 1 image data... masking image... reading psf... ERROR: LoadError: ArgumentError: invalid argument #2306780536149901312 to LAPACK call
 [inlined code] from ./linalg/lapack.jl:27
 in getrf!(::Array{Float64,2}) at ./linalg/lapack.jl:542
 in lufact!(::Array{Float64,2}, ::Type{Val{true}}) at ./linalg/lu.jl:19
 [inlined code] from ./linalg/generic.jl:534
 in evaluate_psf_at_point(::Array{Float64,1}, ::Array{Array{Float64,1},1}, ::Array{Array{Float64,2},1}, ::Array{Float64,1}) at /global/homes/k/kpamnany/.julia/v0.5/SloanDigitalSkySurvey/src/PSF.jl:107
 [inlined code] from ./boot.jl:333
 in render_psf(::Array{Float64,1}, ::Array{Array{Float64,1},2}) at /global/homes/k/kpamnany/.julia/v0.5/SloanDigitalSkySurvey/src/PSF.jl:134
 in (::SloanDigitalSkySurvey.PSF.#evaluate_fit#2{Bool,Array{Float64,2}})(::Array{Float64,1}) at /global/homes/k/kpamnany/.julia/v0.5/SloanDigitalSkySurvey/src/PSF.jl:186
 [inlined code] from ./multidimensional.jl:212
 in #nelder_mead#48(::Float64, ::Float64, ::Float64, ::Float64, ::Array{Float64,1}, ::Int64, ::Bool, ::Bool, ::Void, ::Int64, ::Bool, ::Optim.#nelder_mead, ::SloanDigitalSkySurvey.PSF.#evaluate_fit#2{Bool,Array{Float64,2}}, ::Array{Float64,1}) at /global/homes/k/kpamnany/.julia/v0.5/Optim/src/nelder_mead.jl:74
 [inlined code] from ./boot.jl:331
 in #optimize#56(::Symbol, ::Float64, ::Float64, ::Float64, ::Int64, ::Bool, ::Bool, ::Bool, ::Void, ::Int64, ::Optim.#hz_linesearch!, ::Bool, ::Void, ::Optim.#optimize, ::SloanDigitalSkySurvey.PSF.#evaluate_fit#2{Bool,Array{Float64,2}}, ::Array{Float64,1}) at /global/homes/k/kpamnany/.julia/v0.5/Optim/src/optimize.jl:482
 [inlined code] from ./boot.jl:331
 in #fit_psf_gaussians_least_squares#1(::Array{Float64,1}, ::Float64, ::Int64, ::Bool, ::Int64, ::Symbol, ::Any, ::Array{Float64,2}) at /global/homes/k/kpamnany/.julia/v0.5/SloanDigitalSkySurvey/src/PSF.jl:199
 [inlined code] from ./boot.jl:331
 in fit_raw_psf_for_celeste(::Array{Float64,2}) at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/SkyImages.jl:481
 in #load_sdss_blob#5(::Set{ASCIIString}, ::Any, ::UTF8String, ::ASCIIString, ::ASCIIString, ::ASCIIString) at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/SkyImages.jl:420
 [inlined code] from ./set.jl:10
 in infer(::UTF8String, ::ASCIIString, ::ASCIIString, ::ASCIIString, ::UTF8String, ::Int64, ::Int64) at /global/homes/k/kpamnany/.julia/v0.5/Celeste/src/api.jl:62
 in main() at /global/u1/k/kpamnany/.julia/v0.5/Celeste/bin/celeste.jl:71
 in include(::ASCIIString) at ./boot.jl:264
 in include_from_node1(::UTF8String) at ./loading.jl:417
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
```

**Comment by rgiordan** [comment]
If you're getting the same error, then you're not running the new version
of the script that I sent you, since it doesn't use a Cholseky
factorization any more.  Please pull debug_for_kiran and try again?

**Comment by kpamnany** [comment]
Not the same error, sorry, I updated my note above.

**Comment by kpamnany** [comment]
Let me try again, to be clear.

The debug script ran fine, it's output is in the gist I linked above. Jeff suggested that possibly the problem was the Julia version we were using because he successfully ran celeste.jl with the Julia nightly binary. I rebuilt current Julia master and ran it. Unfortunately, I do see an error; the one above; which is different than the original error.

Hope this is clearer now!

**Comment by rgiordan** [comment]
When I follow the gist link, I see a cholesky error.  Could you please
pull, run the debugging script, and send the full new error?

**Comment by kpamnany** [comment]
Pulled, ran, see the same cholesky error -- look at the gist comments.

**Comment by rgiordan** [comment]
Crap, I'm sorry, I forgot to remove the PSF prefix.  Try again, please?

**Comment by kpamnany** [comment]
https://gist.github.com/kpamnany/d53acc1b08531f64f8e9#file-gistfile1-txt

**Comment by rgiordan** [comment]
That's in the middle of what should be a pretty benign function.  Could you
just run this for me to make sure basic LAPACK is working?

sigma_vec = Array(Matrix{Float64}, 2)
sigma_vec[1] = Float64[2 0; 0 2]
sigma_vec[2] = Float64[3 0; 0 3]

z = Float64[3, 4]

for k = 1:2
  println(dot(z, sigma_vec[k] \ z))
  println(logdet(sigma_vec[k]))
end

**Comment by kpamnany** [comment]
```
julia> sigma_vec = Array(Matrix{Float64}, 2)

2-element Array{Array{Float64,2},1}:
 #undef
 #undef

julia> sigma_vec[1] = Float64[2 0; 0 2]
2x2 Array{Float64,2}:
 2.0  0.0
 0.0  2.0

julia> sigma_vec[2] = Float64[3 0; 0 3]
2x2 Array{Float64,2}:
 3.0  0.0
 0.0  3.0

julia> z = Float64[3, 4]
2-element Array{Float64,1}:
 3.0
 4.0

julia> for k = 1:2
       println(dot(z, sigma_vec[k] \ z))
       println(logdet(sigma_vec[k]))
       end
12.5
1.3862943611198906
8.333333333333332
2.1972245773362196
```
